src/Resources/Resources/Models.Authorization/AuthorizationClient.cs

```diff
@@ -209,15 +210,35 @@ public List<PSRoleAssignment> FilterRoleAssignments(FilterRoleAssignmentsOptions
                     }
 
                     principalId = adObject.Id.ToString();
-                    odataQuery = new ODataQuery<RoleAssignmentFilter>(f => f.AssignedTo(principalId));
+                    if (!hasScope)
+                    {
+                        odataQuery = new ODataQuery<RoleAssignmentFilter>(f => f.AssignedTo(principalId));
+                    }
+                    else
+                    {
+                        odataQuery = new ODataQuery<RoleAssignmentFilter>(f => f.AtScope() && f.AssignedTo(principalId));
+                    }
                 }
                 else
                 {
                     principalId = string.IsNullOrEmpty(options.ADObjectFilter.Id) ? adObject.Id.ToString() : options.ADObjectFilter.Id;
-                    odataQuery = new ODataQuery<RoleAssignmentFilter>(f => f.PrincipalId == principalId);
+                    if (!hasScope)
+                    {
+                        odataQuery = new ODataQuery<RoleAssignmentFilter>(f => f.PrincipalId == principalId);
+                    } 
+                    else
+                    {
+                        // https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments-list-rest
+                        odataQuery = new ODataQuery<RoleAssignmentFilter>(f => f.AtScope() && f.AssignedTo(principalId));
```

Comment:
Previous query:
//subscriptions/4004a9fd-d58e-48dc-aeb2-4a4aec58606f/resourceGroups/PowershellTest/providers/Microsoft.Authorization/roleAssignments?$filter=principalId%20eq%20'f2dc21ac-702a-4bde-a4ce-146edf751d81'&api-version=2020-08-01-preview",
Current query:
Unable to find a matching HTTP request for URL 'GET //subscriptions/4004a9fd-d58e-48dc-aeb2-4a4aec58606f/resourceGroups/PowershellTest/providers/Microsoft.Authorization/roleAssignments?$filter=atScope()%20and%20assignedTo('f2dc21ac-702a-4bde-a4ce-146edf751d81')'. Calling method Item().

Please check this line whether is correct way to write query.

Reply:
It's resolved, filter condition uses '+' as separator